Confusing error and panics when protoc isn't installed.

I didn't have protoc installed on my path, or the protoc include libraries in any place they could be found. I received the following when I ran `gunk generate` and `go test`. After installing protoc correctly everything worked as expected.

```
$ cd examples/util && gunk generate .
2018/11/19 12:50:48 protoc-gen-go: error:can't find object with type .google.protobuf.Empty
error: error executing protoc-gen-go: , exit status 1
```

```
$ go test
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5c020a]

goroutine 1 [running]:
github.com/golang/protobuf/protoc-gen-go/generator.(*Generator).ObjectNamed(0xc000080480, 0xc00001c2d0, 0x22, 0xc0000ed480, 0xc0001f2510)
        /home/jonathan/go/pkg/mod/github.com/golang/protobuf@v1.2.0/protoc-gen-go/generator/generator.go:949 +0x45a
github.com/golang/protobuf/protoc-gen-go/generator.(*Generator).GoType(0xc000080480, 0xc0000ce280, 0xc0000cab60, 0xc00000b320, 0x2, 0x2, 0xa)
        /home/jonathan/go/pkg/mod/github.com/golang/protobuf@v1.2.0/protoc-gen-go/generator/generator.go:1633 +0x49d
github.com/golang/protobuf/protoc-gen-go/generator.(*Generator).generateMessage(0xc000080480, 0xc0000ce280)
        /home/jonathan/go/pkg/mod/github.com/golang/protobuf@v1.2.0/protoc-gen-go/generator/generator.go:2498 +0x444
github.com/golang/protobuf/protoc-gen-go/generator.(*Generator).generate(0xc000080480, 0xc00007ea50)
        /home/jonathan/go/pkg/mod/github.com/golang/protobuf@v1.2.0/protoc-gen-go/generator/generator.go:1149 +0x5bf
github.com/golang/protobuf/protoc-gen-go/generator.(*Generator).GenerateAllFiles(0xc000080480)
        /home/jonathan/go/pkg/mod/github.com/golang/protobuf@v1.2.0/protoc-gen-go/generator/generator.go:1093 +0x27e
main.main()
        /home/jonathan/go/pkg/mod/github.com/golang/protobuf@v1.2.0/protoc-gen-go/main.go:87 +0x1c7
--- FAIL: TestGenerate (0.08s)
    main_test.go:60: error executing protoc-gen-go: , exit status 2
FAIL
exit status 1
FAIL    github.com/gunk/gunk    0.230s
```

I am unsure why we didn't error earlier (when using protoc to load imports), but we should add a `protoc` check that checks it is installed, and using the correct version.

--- a/generate/generate.go
+++ b/generate/generate.go
@@ -850,7 +850,7 @@ func (g *Generator) loadProtoDeps() error {
 
 	files, err := loader.LoadProto(list...)
 	if err != nil {
-		return nil
+		return err
 	}
 
 	for _, pfile := range files {

--- a/loader/loader.go
+++ b/loader/loader.go
@@ -178,7 +178,7 @@ syntax = "proto3";
 	out, err := cmd.Output()
 	if err != nil {
 		if e, ok := err.(*exec.ExitError); ok {
-			return nil, fmt.Errorf("%s", e.Stderr)
+			return nil, fmt.Errorf("protoc %s: %s", e, e.Stderr)
 		}
 		return nil, err
 	}

--- a/testdata/scripts/envs.txt
+++ b/testdata/scripts/envs.txt
@@ -1,0 +1,30 @@
+env HOME=$WORK/home
+
+# protoc can't be run correctly to resolve proto imports
+env PATH=$WORK/bin:$PATH
+exec chmod a+x bin/protoc
+
+# we should stop before any generators are run
+! gunk generate .
+! stdout .
+stderr 'exit status'
+stderr 'fatal failure'
+! stderr 'error executing protoc-gen'
+! stderr panicc
+
+-- bin/protoc --
+#!/bin/sh
+
+echo fatal failure >&2
+
+exit 1
+-- go.mod --
+module testdata.tld/util
+-- doc.go --
+package util // make this directory a Go package
+-- echo.gunk --
+package util // proto "testdata.v1.util"
+
+type Util interface {
+	Echo() // use google.protobuf.Empty, which requires protoc
+}

